homeassistant/components/xiaomi_miio/switch.py

```diff
@@ -129,6 +136,26 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the switch from a config entry."""
     entities = []
 
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_GATEWAY:
+        gateway = hass.data[DOMAIN][config_entry.entry_id][CONF_GATEWAY]
+        # Gateway sub devices
+        sub_devices = gateway.devices
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        for sub_device in sub_devices.values():
+            if sub_device.device_type == "Switch":
+                if "status_ch0" in sub_device.status.keys():
```

Comment:
Maybe create a constant list with valid data keys instead, and then check if any of those valid keys are in the device status, and pass the data key instead of the integer to the entity.

Reply:
I changed it to make it more general and easier to pass aditional info like icons in the future